### PR TITLE
Refactor MapLoaderDaoFileBased, MapSetThumbnailDaoImpl, RouteDaoImpl

### PR DESCRIPTION
**Pull request by theycome**
As suggested by 
https://github.com/p-lr/TrekMe/pull/95#discussion_r1421317545

### CODE REVIEW

**Comment by theycome** [comment]
in `core.map.data.dao` package.

File: app/src/main/java/com/peterlaurence/trekme/core/map/data/dao/MapLoaderDaoFileBased.kt [hunk 1]
```diff
@@ -111,9 +108,9 @@ class MapLoaderDaoFileBased constructor(
                  * properly). */
                 val shouldSaveUUID = mapGson.uuid == null
 
-                val thumbnailImage = if (mapGson.thumbnail != null) {
```
Comment (p-lr) [inline app/src/main/java/com/peterlaurence/trekme/core/map/data/dao/MapLoaderDaoFileBased.kt:114]: In this particular case I find the `if` expression clearer.

File: app/src/main/java/com/peterlaurence/trekme/core/map/data/dao/MapSetThumbnailDaoImpl.kt [hunk 2]
```diff
@@ -32,7 +32,7 @@ class MapSetThumbnailDaoImpl(
             makeThumbnail(uri, contentResolver, map.thumbnailSize, imageOutputStream)
         }
 
-        return if (thumbnailImage != null) {
+        return thumbnailImage?.let {
```
Comment (p-lr) [inline app/src/main/java/com/peterlaurence/trekme/core/map/data/dao/MapSetThumbnailDaoImpl.kt:35]: In this particular case I find the `if` expression clearer

**Comment by p-lr** [comment]
To sum up, in theory in Kotlin we could replace every single `if` making null comparisons by a couple of `let` or `also` calls, along with `?:` operator. However _in simple cases_, plain old `if`s read better in my experience.

**Comment by p-lr** [comment]
I squashed into one commit, thanks.

**Pull request merged**